### PR TITLE
Allow weighted scalar reporting by providing a tuple to the reporter.…

### PR DESCRIPTION
**Pull request by hknerdgn**
Allow weighted scalar reporting by providing a tuple to the reporter. compute_mean computes weighted mean.

Relates to issue https://github.com/chainer/chainer/issues/4004 .

### CODE REVIEW

**Comment by kmaehashi** [comment]
@hknerdgn ping?

**Comment by hknerdgn** [comment]
Can someone take over? I do not have time unfortunately. It is a simple PR anyway.

**Comment by mitmul** [comment]
@hknerdgn Thanks! I hand over this.

File: chainer/reporter.py [hunk 1]
```diff
@@ -314,15 +317,24 @@ def add(self, d):
...
         summaries = self._summaries
         for k, v in six.iteritems(d):
+            w = 1
+            if isinstance(v, tuple):
+                w = v[1]
+                v = v[0]
+                if isinstance(w, variable.Variable):
+                    w = w.data
+                if not numpy.isscalar(w) and not getattr(w, 'ndim', -1) == 0:
+                    w = 1
```
[the remark this answers is not in the document]
Comment (mitmul) [reply chainer/reporter.py]: Thanks, I fixed this in #4844

File: chainer/reporter.py [hunk 2]
```diff
@@ -314,15 +317,24 @@ def add(self, d):
         Args:
             d (dict): Dictionary of scalars to accumulate. Only elements of
                scalars, zero-dimensional arrays, and variables of
-               zero-dimensional arrays are accumulated.
+               zero-dimensional arrays are accumulated. When the value
+               is a tuple, the second is interpreted as a weight.
```
[the remark this answers is not in the document]
Comment (mitmul) [reply chainer/reporter.py]: Thanks, I fixed this in #4844

**Pull request closed without merging**